### PR TITLE
feat: CU-1n5w3vb add loading and variants to badge

### PR DESCRIPTION
**Pull request by snqb**
#### Card: https://app.clickup.com/t/1n5w3vb

This PR adds:
- the awaited loading state for the badge
- bordered `mode` 
- `icon` prop
- minor changes to the color scheme

### BREAKING:

this PR removes these values from `variant` prop:
- `bold`, which is replaced with `default` now
- `border`, we now use the `mode="bordered"`

#### To be noted:
- the `warning` variant colors are changed, but this should not lead to changes to the code, as the semantics stay the same

#### DEMO:

https://user-images.githubusercontent.com/19791699/144876089-7429d467-b72c-4389-923f-6a98b2f9b371.mp4

### CODE REVIEW

**Comment by giardiv** [comment]
Task linked: [CU-1n5w3vb Badges: update and clean code](https://app.clickup.com/t/1n5w3vb)

File: src/components/badges/badge.tsx [hunk 1]
```diff
@@ -1,41 +1,103 @@
...
 export interface BadgeProps extends Omit<FlexProps, 'css'> {
   value: string | number;
-  variant?: 'light' | 'bold' | 'fail' | 'warning' | 'success' | 'label' | 'border';
+  variant?: Variant;
+  mode?: Mode;
+  loading?: boolean;
+  icon?: IconName;
 }
 
+// this is a workaround because styled-system does not support theme colors for `linear-gradient`
```
Comment (snqb) [inline src/components/badges/badge.tsx:27]: https://github.com/rebassjs/rebass/issues/1059

File: src/theme/theme.ts [hunk 2]
```diff
@@ -69,7 +69,8 @@ const theme: ITheme = {
       red: '#EB5757',
       orange: '#f2994a',
       purple: '#9B51E0',
-      yellow: '#FFF066',
+      yellow: '#FFE600',
```
Comment (snqb) [inline src/theme/theme.ts:72]: this yellow was just wrong, see [Figma](https://www.figma.com/file/mvY1f3xMXUEd6szvT82FxB/Hopsworks-UI?node-id=9288%3A804)

File: src/theme/badges/primary.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const variants = {
-  bold: {
+  default: {
     '> span': {
-      color: 'white',
+      color: 'grayShade3',
```
Comment (snqb) [inline src/theme/badges/primary.ts:4]: the base color was changed

**Review by giardiv — APPROVED** [review]

**Review by ErmiasG — APPROVED** [review]

**Pull request merged**